tsdb/head.go

```diff
@@ -341,6 +352,17 @@ func (h *Head) resetInMemoryState() error {
 	return nil
 }
 
+func (h *Head) resetWLReplayResources() {
```

Comment:
I doubt this is needed. In fact I suspect this actually only increase allocs and the GC is removing objs from inside pools in the next GC iteration (due to victim area implementation mentioned in https://victoriametrics.com/blog/go-sync-pool/). 

Unless we can prove it e.g. with some experiment, I think I would remove this.

Reply:
You're right, I was missing the second GC, benchmark's `memprofile` only runs one.